run.py

```diff
@@ -4,260 +4,71 @@
-  if base_model is None:
-    raise ValueError('base_config class has to be defined in the config file')
-
-  # after we read the config, trying to overwrite some of the properties
-  # with command line arguments that were passed to the script
-  parser_unk = argparse.ArgumentParser()
-  for pm, value in flatten_dict(base_config).items():
-    if type(value) == int or type(value) == float or \
-       isinstance(value, string_types):
-      parser_unk.add_argument('--' + pm, default=value, type=type(value))
-    elif type(value) == bool:
-      parser_unk.add_argument('--' + pm, default=value, type=ast.literal_eval)
-  config_update = parser_unk.parse_args(unknown)
-  nested_update(base_config, nest_dict(vars(config_update)))
-
-  # checking that everything is correct with log directory
-  logdir = base_config['logdir']
-  if args.benchmark:
-    args.no_dir_check = True
-  try:
-    if args.enable_logs:
-      ckpt_dir = os.path.join(logdir, 'logs')
-    else:
-      ckpt_dir = logdir
-    if args.mode == 'train' or args.mode == 'train_eval':
-      if os.path.isfile(logdir):
-        raise IOError("There is a file with the same name as \"logdir\" "
-                      "parameter. You should change the log directory path "
-                      "or delete the file to continue.")
-
-      # check if "logdir" directory exists and non-empty
-      if os.path.isdir(logdir) and os.listdir(logdir) != []:
-        if not args.continue_learning:
-          raise IOError("Log directory is not empty. If you want to continue "
-                        "learning, you should provide "
-                        "\"--continue_learning\" flag")
-        checkpoint = tf.train.latest_checkpoint(ckpt_dir)
-        if checkpoint is None:
-          raise IOError(
-              "There is no valid TensorFlow checkpoint in the "
-              "{} directory. Can't load model".format(ckpt_dir)
-          )
-      else:
-        if args.continue_learning:
-          raise IOError("The log directory is empty or does not exist. "
-                        "You should probably not provide "
-                        "\"--continue_learning\" flag?")
-        checkpoint = None
-    elif args.mode == 'infer' or args.mode == 'eval':
-      if os.path.isdir(logdir) and os.listdir(logdir) != []:
-        checkpoint = tf.train.latest_checkpoint(ckpt_dir)
-        if checkpoint is None:
-          raise IOError(
-              "There is no valid TensorFlow checkpoint in the "
-              "{} directory. Can't load model".format(ckpt_dir)
-          )
-      else:
-        raise IOError(
-            "{} does not exist or is empty, can't restore model".format(
-                ckpt_dir
-            )
-        )
-  except IOError as e:
-    if args.no_dir_check:
-      print("Warning: {}".format(e))
-      print("Resuming operation since no_dir_check argument was provided")
-    else:
-      raise
+  if args.mode == "interactive_infer":
+    deco_print(
```

Comment:
Why is this not an exception then? Or maybe we should just blacklist this mode in the get_base_config? It will also save us from doing additional checks later in this script.